libsoftwarecontainer/src/gateway/devicenode/devicenodegateway.cpp

```diff
@@ -51,38 +51,45 @@ bool DeviceNodeGateway::activateGateway()
     }
 
     for (auto &dev : devlist) {
-        log_info() << "Mapping device " << dev.name;
+        if (!dev.isConfigured) {
+            log_info() << "Mapping device " << dev.name;
 
-        std::string devicePathInContainerOnHost = buildPath(getContainer()->rootFS(), dev.name);
-        std::string deviceParent = parentPath(devicePathInContainerOnHost);
+            std::string devicePathInContainerOnHost = buildPath(getContainer()->rootFS(), dev.name);
+            std::string deviceParent = parentPath(devicePathInContainerOnHost);
```

Comment:
I just realized that this whole check for `deviceParent` etc only works because we are lucky. The /dev/ directory inside the container is mounted as a tmpfs, and as such, we can't see it from the host side.

@kursatkobya can you update the test to try to mount `/dev/input/event0` into the container? I suspect this should fail (I should've taken care of this when I handled the general bind-mounting issue) because `/dev/input` does not exist inside the container, and the `createDirectory` calls will only create it on the host (not inside the tmpfs where `/dev` in the container resides).

If that is the case, we should update to try to create any needed parent directories inside the container (with a functionjob, I guess).